### PR TITLE
feat(mixins): add withLayout mixin

### PR DESCRIPTION
**Pull request by THoj13**
Adds `withLayout` mixins.

To better display the effects of this mixin, the storybook should be updated to use `withMetaData` and `MetadataTile` once those have been added to this repo.

### CODE REVIEW

File: mixins/withLayout/withLayout.mdx [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+
+Use `withLayout` to autotomatically size an item according to screen dimensions, upCount and/or set explicitly.
+
+## Usage
+
+Let's use an example to see how we can use `withLayout`. Let's say I have an a `Tile` component on screen that should be sized according the screen size:
+
+```js
+import lng from '@lightningjs/core';
+import { withLayout } from '@lightningjs/mixins';
```
Comment (anthony9187) [inline mixins/withLayout/withLayout.mdx]: lets update this to reflect the correct package name

File: mixins/withLayout/withLayout.mdx [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+    };
+  }
+}
+```
+
+If I want to explicitly set a height but the only other information is the screen dimensions, I can do that by adding these parameters to the itemLayout object.
+
+```js
+import lng from '@lightningjs/core';
+import { withLayout } from '@lightningjs/mixins';
```
Comment (anthony9187) [inline mixins/withLayout/withLayout.mdx]: same here

File: mixins/withLayout/withLayout.mdx [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+    };
+  }
+}
+```
+
+If I am unsure of the width, I can autosize by adding a height and upCount properties to the itemLayout object.
+
+```js
+import lng from '@lightningjs/core';
+import { withLayout } from '@lightningjs/mixins';
```
Comment (anthony9187) [inline mixins/withLayout/withLayout.mdx]: aaaaaand one more

**Review by anthony9187 — APPROVED** [review]
just a couple small changes, looks good!

**Review by anthony9187 — APPROVED** [review]
looks good!

**Review by ImCoolNowRight — APPROVED** [review]

**Pull request merged**